src/KubernetesClientConfiguration.InCluster.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using k8s.Exceptions;
+
+namespace k8s
+{
+    public partial class KubernetesClientConfiguration
+    {
+        private const string ServiceAccountTokenKey = "token";
+        private const string ServiceAccountRootCAKey = "ca.crt";
+
+        public static KubernetesClientConfiguration InClusterConfig()
+        {
+            var host = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST");
+            var port = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_PORT");
+
+            if (string.IsNullOrWhiteSpace(host) || string.IsNullOrWhiteSpace(port))
+            {
+                throw new KubeConfigException(
+                    "unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined");
+            }
+
+            var token = File.ReadAllText("/var/run/secrets/kubernetes.io/serviceaccount/" + ServiceAccountTokenKey);

```

Comment:
extract `"/var/run/secrets/kubernetes.io/serviceaccount/"` and other strings out as constants?

Reply:
another small question
Is this still not supported by windows after 1709?